streaming/src/main/scala/org/apache/spark/streaming/receiver/ReceivedBlockHandler.scala

```diff
@@ -33,6 +33,7 @@ import org.apache.spark.{Logging, SparkConf, SparkException}
 /** Trait that represents the metadata related to storage of blocks */
 private[streaming] trait ReceivedBlockStoreResult {
   def blockId: StreamBlockId  // Any implementation of this trait will store a block id
+  def numRecords: Long        // Any implementation of this trait will store the number of records
```

Comment:
"will have to return the number of records"

Also, this should probably be an Option. See the #6659 that I merged. Please merge with master to get those changes.